File: selection_sort.js
function selectionSort(arr) {
    for (let i = 0; i < arr.length; i++) {
      let lowest = i
      for (let j = i + 1; j < arr.length; j++) {
        if (arr[j] < arr[lowest]) {
          lowest = j
        }
      }
      if (lowest !== i) {
        // Swap
        ;[arr[i], arr[lowest]] = [arr[lowest], arr[i]]
      }
      console.log("It Swaped: ", arr[i], arr[lowest])
      console.log(arr)
    }
    return arr
  }
  let array = [4,2,-3,-17,7]
  console.log(array)
  console.log(selectionSort(array))

File: selection_sort.py
# Selection sort
# time complexity O(n^2)
import random
def selectionSort(array, size):
    for ind in range(size):
        min_index = ind
        for j in range(ind + 1, size):
			# select the minimum element in every iteration
            if array[j] < array[min_index]:
                min_index = j
        print("It Swaped: ", array[ind], array[min_index])
        array[ind], array[min_index] = array[min_index], array[ind]
        print(array)

arr = random.sample(range(-10, 30), 10)
print("Input Array: ",arr,"\n")
size = len(arr)
selectionSort(arr, size)
